### PR TITLE
feat: Generate ESLint config files

### PR DESCRIPTION
**Pull request by markdalgleish**
To support ESLint editor integrations, we now generate config files in the root of the project directory.

Similar to Prettier, we now run ESLint across the whole project, ignoring files with `.eslintignore` which is generated automatically.

Any modifications to the ESLint config in `dangerouslySetESLintConfig` are written to disk as part of your `.eslintrc`.

### CODE REVIEW

File: lib/configure.js [hunk 1]
```diff
@@ -54,6 +55,12 @@ module.exports = async () => {
     gitIgnorePatterns.push(tslintConfigFileName);
   }
 
+  // Generate ESLint configuration
+  const eslintConfigFilename = '.eslintrc';
+  console.log('eslintConfig', eslintConfig);
```
Comment (mattsjones) [inline lib/configure.js]: Console.log slip through ?

Comment (markdalgleish) [reply lib/configure.js]: Oops. Yep.

Comment (markdalgleish) [reply lib/configure.js]: Fixed 👍

File: test/test-cases/typescript-css-modules/app/sku.config.js [hunk 2]
```diff
@@ -3,5 +3,13 @@ module.exports = {
     client: 'src/client.tsx',
     render: 'src/render.tsx',
     server: 'src/server.tsx'
+  },
+  // Use 'dangerouslySetESLintConfig' so we can
+  // check that overrides are written to disk.
```
Comment (michaeltaranto) [inline test/test-cases/typescript-css-modules/app/sku.config.js]: Is there a corresponding test for the written output?

File: lib/configure.js [hunk 3]
```diff
@@ -54,6 +55,11 @@ module.exports = async () => {
     gitIgnorePatterns.push(tslintConfigFileName);
   }
 
+  // Generate ESLint configuration
+  const eslintConfigFilename = '.eslintrc';
+  await writeFileToCWD(eslintConfigFilename, eslintConfig);
```
Comment (michaeltaranto) [inline lib/configure.js]: Confirming, can `.eslintrc` handle the comments in json? By default the comment banner is prepended so just making sure.

Comment (markdalgleish) [reply lib/configure.js]: Yep, it can.

File: test/test-cases/typescript-css-modules/app/sku.config.js (hunk 2 above)
Comment (markdalgleish) [reply test/test-cases/typescript-css-modules/app/sku.config.js]: Just added a test 👍

Comment (markdalgleish) [reply test/test-cases/typescript-css-modules/app/sku.config.js]: Let me know what you think.

**Pull request merged**